tool/lib/commands/fix_goldens.dart

```diff
@@ -0,0 +1,101 @@
+      'run-id',
+      help: 'The ID of the workflow run where the goldens are failing. '
+          'e.g.https://github.com/flutter/devtools/actions/runs/<run-id>/job/16691428186',
+      valueHelp: '12345',
+      mandatory: true,
+    );
+  }
+  @override
+  String get description =>
+      'A helper script for downloading and applying golden fixes, when they are broken.';
+
+  @override
+  String get name => 'fix-goldens';
+
+  final processManager = ProcessManager();
+
+  @override
+  FutureOr? run() async {
+    // Change the CWD to the repo root
+    Directory.current = pathFromRepoRoot("");
+
+    final runId = argResults!['run-id']!;
+    final tmpDownloadDir = Directory(
+      path.join('.tmp${DateTime.now().millisecondsSinceEpoch}'),
+    );
+    tmpDownloadDir.createSync();
+    try {
+      print('Downloading the artifacts to ${tmpDownloadDir.path}');
+      await processManager.runProcess(
+        CliCommand.from('gh', [
+          'run',
+          'download',
+          runId,
+          '-p',
+          '*golden_image_failures*',
+          '-D',
+          tmpDownloadDir.path,
+        ]),
+      );
+
+      final downloadedGoldens = tmpDownloadDir
+          .listSync(recursive: true)
+          .where((e) => e.path.endsWith('testImage.png'));
+      final allLocalGoldenPngs =
+          Directory(pathFromRepoRoot("packages/devtools_app/test/"))
+              .listSync(recursive: true)
+            ..where((e) => e.path.endsWith('.png'));
+
+      for (final downloadedGolden in downloadedGoldens) {
+        final downloadedGoldenBaseName = path.basename(downloadedGolden.path);
+        final expectedGoldenFileName =
+            '${RegExp(r'(^.*)_testImage.png').firstMatch(downloadedGoldenBaseName)?.group(1)}.png';
+
+        final fileMatches = allLocalGoldenPngs.where(
+          (e) => e.path.endsWith(expectedGoldenFileName),
+        );
+
+        final String destinationPath;
+        if (fileMatches.isEmpty) {
+          throw 'Could not find a golden Image for $downloadedGoldenBaseName using $expectedGoldenFileName as '
+              'the name of the search.';
+        } else if (fileMatches.length == 1) {
+          destinationPath = fileMatches.first.path;
+        } else {
+          print("Multiple goldens found for ${downloadedGolden.path}");
+          print("Select which golden should be overridden:");
+
+          for (int i = 0; i < fileMatches.length; i++) {
+            final fileMatch = fileMatches.elementAt(i);
+            print('${i + 1}) ${fileMatch.path}');
+          }
+
+          final userSelection = int.parse(stdin.readLineSync()!);
+
+          destinationPath = fileMatches.elementAt(userSelection - 1).path;
+        }
+        await downloadedGolden.rename(destinationPath);
+
+        print("Fixed: $destinationPath");
+      }
+
+      print('Done updating ${downloadedGoldens.length} goldens');
+    } finally {
+      tmpDownloadDir.deleteSync(recursive: true);
+    }
```

Comment:
I think you need to call

```
    // Closes stdin for the entire program.
    await sharedStdIn.terminate();
```
at the end or this script will not terminate due to an open port

Reply:
I haven't had any issues with this yet. What makes you think the stdin would hang?